src/Common/tests/System/Net/RemoteServerQuery.cs

```diff
@@ -0,0 +1,79 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Net.Test.Common
+{
+    internal static class RemoteServerQuery
+    {
+        internal static async Task Run(Func<Task> testCode, Type remoteServerFailExceptionType, string servername, bool mustFail = true)
+        {
+            try
+            {
+                await testCode();
+            }
+            catch (Exception actualException)
+            {
+                if (remoteServerFailExceptionType.Equals(actualException.GetType()))
+                {
+                    if (mustFail)
+                        Assert.True(false, $"External issue with remote server: {servername}");
```

Comment:
I would prefix the text with "Likely" -> "Likely external issue with remote server: ..."

Reply:
ok